removed file dictionary, and use full path for files in existing code…

pbharrin opened the pull request:
This PR is a cleanup of the Existing Code PR.  In the original implementation there was a dictionary with file names as keys and file full paths as values.  For example: 
main.py --> /Users/boss/gpt-engineer/projects/example/archive/20230816_103041/workspace/main.py

The problem with this is that there could be collision in the dict if files in different folders have the same name.  (For example util.py in multiple folders could cause a collision).  This dict has been completely removed and now the full paths are passed to the AI and returned.

AntonOsika commented on gpt_engineer/chat_to_files.py:
```diff
@@ -51,21 +51,12 @@ def overwrite_files(chat, dbs, replace_files):
             dbs.workspace[file_name] = file_content
+        else:
+            with open(file_name, "w") as text_file:
+                text_file.write(file_content)
```
Doesn't this do exactly the same thing as `dbs.workspace[file_name] = file_content`?

pbharrin replied:
yes, it does.  fixed

AntonOsika commented on gpt_engineer/file_selector.py:
```diff
@@ -187,7 +188,8 @@ def ask_for_selection(self) -> List[str]:
                 user_input = (
-                    user_input.replace("", ",") if " " in user_input else user_input
+                    user_input.replace(
+                        "", ",") if " " in user_input else user_input
```
Not sure what this is for but shouldn't it be this:

```suggestion
                        " ", ",") if " " in user_input else user_input
```

pbharrin replied:
Fixed.

AntonOsika commented on gpt_engineer/file_selector.py:
```diff
@@ -196,10 +198,12 @@ def ask_for_selection(self) -> List[str]:
                         for num in range(start, end + 1):
-                            selected_paths.append(str(self.selectable_file_paths[num]))
+                            selected_paths.append(
+                                str(self.selectable_file_paths[num]))
```
I think this makes the CI break.

Have you configured your pre-commit hooks like described [here](https://github.com/AntonOsika/gpt-engineer/blob/main/.github/CONTRIBUTING.md)?

I think just

`pre-commit install`

AntonOsika reviewed (changes requested):
Looks good, I had tw nitpicks though wanna have a look before merging?

AntonOsika commented:
You can see CI being red above below:

pbharrin commented:
The security issue has been addressed, now files paths are sent over relative to the working directory.  I believe I have addressed everything.  

I fixed the pre-checks, thanks for the info I did not know how to do that.
